### PR TITLE
support "dices" spelling for dice trigger

### LINKED ISSUE #1255: Dice: Support for "Roll 2 dices" or "Roll two dice"
On the instant answer "dice":

Keyword: "roll 2 dice" works, although I'm seeing blank rectangles but that's supposedly my browser unicode support that is lacking (latest chrome).

image

I would have supported ***"roll 2 dices"*** too. Dices is an accepted word of the [dictionary](http://www.thefreedictionary.com/dices), albeit less common than dice.

***"Roll two dice"*** doesn't work either.

-----
IA Page: https://duck.co/ia/view/dice

### PR DESCRIPTION
**Pull request by marianosimone**
fixes #1255

I didn't include the part about converting "two" into 2, as I wouldn't know where to stop (one, two, three, ten, twenty???)

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by GuiltyDolphin** [comment]
@marianosimone `dices`? I don't even think that's a word. There's `die` and `dice`.

**Comment by mintsoft** [comment]
@GuiltyDolphin True, but I imagine that people will search for it

**Comment by GuiltyDolphin** [comment]
@mintsoft Fair dos. I'm not sure how well it will play with Numericalize; @marianosimone you might want to isolate the number-words from the rest of the query so Numericalize doesn't mess it up. It might be fine though seeing as it will just strip the extra 's'.

**Comment by marianosimone** [comment]
@GuiltyDolphin @mintsoft thanks for pointing out `Numericalize`. Already integrated!

BTW, as #1255 indicates, `dices` is an accepted form (according to http://www.thefreedictionary.com/dices)

**Comment by GuiltyDolphin** [comment]
@marianosimone Yeah, it's fine.

Can we add a test or two showing the combination of word-numbers with `dices`?

**Comment by GuiltyDolphin** [comment]
@marianosimone Just a couple using the words (like `three`, `seven` etc...) along with `dices`, to make sure they play nicely.

File: t/Dice.t [hunk 1]
```diff
@@ -91,6 +105,25 @@ ddg_goodie_test(
             }
        }
     ),
+
+    # Query with numbers as words
+    "roll five dice" => test_zci(
```
Comment (marianosimone) [inline t/Dice.t]: @GuiltyDolphin you mean this?

Comment (GuiltyDolphin) [inline t/Dice.t]: No, I mean combining both the things you are adding - `dices` and the number-word triggers. Like `roll seven dices`.

**Comment by marianosimone** [comment]
@GuiltyDolphin added!

File: lib/DDG/Goodie/Dice.pm [hunk 2]
```diff
@@ -76,12 +80,16 @@ handle remainder_lc => sub {
     my $heading = "Random Dice Roll";
     my $total; # total of all dice rolls
     foreach (@values) {
-        if ($_ =~ /^(?:a? ?die|(\d{0,2})\s*dic?e)$/) {
-            # ex. 'a die', '2 dice', '5dice'
+        if ($_ =~ /^(?:a? ?die|(\d{0,2}|[a-zA-Z]+)\s*dic?es?)$/) {
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Dice.pm]: This won't catch all numbers - there are also phrases like `twenty-seven` and `three hundred and fifty-four`. We can probably just pass the stripped-down version off to `Numericalize`.

Comment (marianosimone) [inline lib/DDG/Goodie/Dice.pm]: Mhhh... I'm only using this to catch simple cases (there's a limit on how many dice we accept anyway, which is 99).

I haven't thought about the '-' case though, so maybe supporting `[\s\-]` is a good idea. Adding now!

**Comment by GuiltyDolphin** [comment]
@marianosimone Thanks, this looks great! :dolphin:

**Pull request merged**